recipes/openscenegraph/all/test_package/conanfile.py

```diff
@@ -0,0 +1,25 @@
+from conans import CMake, ConanFile, tools
+import os
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake", "cmake_find_package"
+
+    def build(self):
+        cmake = CMake(self)
+        for key, value in self.options["openscenegraph"].items():
+            if key.startswith("with_"):
+                cmake.definitions["OSG_HAS_" + key.upper()] = 1 if value else 0
+        if self.settings.os == "Macos":
+            cmake.definitions["OSG_HAS_WITH_GIF"] = 0
```

Comment:
Why don't you pass these definitions by cpp_info.defines ?

Reply:
It wouldn't be appropriate to put them anywhere other than in the test package, as they aren't standard.

Here it serves two purposes:
1. To allow us to check that OSG was built with the expected set of plugins, which isn't needed in general.
2. To allow us to load plugins correctly when OSG is built as a static library.

The expected usage of OSG is for an application is to dynamically link to OSG, and then query OSG to see if it supported some functionality, and then gracefully fail if OSG doesn't. This doesn't require any preprocessor mumbo-jumbo as it all is resolved at runtime.

Generally speaking, when developers ask how to use OSG as a static library, the answer is "don't". As I understand it, the only reason why OSG has any support for being used as a static library was to support the needs of some embedded developers. You can easily encounter issues where plugins are multiply loaded, giving linking issues, or other sorts of unexpected behavior. Using OSG as a static library should be reserved for experts who know what they are doing (and thus have their own way of handling OSG plugins).